modules/gapi/CMakeLists.txt

```diff
@@ -369,6 +369,8 @@ if(HAVE_ONNX)
   ocv_target_compile_definitions(${the_module} PRIVATE HAVE_ONNX=1)
   if(HAVE_ONNX_DML)
     ocv_target_compile_definitions(${the_module} PRIVATE HAVE_ONNX_DML=1)
+    # FIXME: It shouldn't be here...
+    ocv_target_compile_definitions(${the_module} PRIVATE HAVE_DXCORE=1)
```

Comment:
@opencv-alalek Could you advice how `dxcore`, `d3d12` and `directml` can be detected by opencv?
Should we use the similar approach as there:
https://github.com/opencv/opencv/blob/4.x/cmake/OpenCVDetectDirectX.cmake ?

Reply:
Yes, we need to have something like `OpenCVDetectDirectML.cmake`

Reply:
Done